### PR TITLE
Fixed username and domain separation in getACL

### PR DESCRIPTION
**Pull request by Snafu**
Inspired by anikrooz commit
https://github.com/nextcloud/server/commit/1f95f9b4791880e8d054b27fd4b9f50a7434cc65

Addresses issue mentioned here: https://github.com/nextcloud/server/issues/26457#issuecomment-822648601

Signed-off-by: Christopher Gabriel <stuff@c-gabriel.at>

### CODE REVIEW

**Comment by Snafu** [comment]
I want to point out the comment by anikrooz in the commit that inspired my fix and for some reason I did not pay enough attention to:
"Returning a plain username eg. ['', $user] could be a security risk or at least would be incorrect".
I'm not an expert regarding ACL security, if the above statement is true, then the original fix should be implemented, returning an empty username.

File: apps/files_external/lib/Lib/Storage/SMB.php [hunk 1]
```diff
@@ -142,7 +142,7 @@ public function __construct($params) {
 	private function splitUser($user) {
 		if (strpos($user, '/')) {
```
Comment (Snafu) [inline apps/files_external/lib/Lib/Storage/SMB.php:143]: I'm not familiar with this ACL user notation containing a slash. Maybe the strpos() check here also has to be changed to `strpos($user, '/') !== false`

**Comment by csware** [comment]
Cross-link: #25540

**Comment by skjnldsv** [comment]
@Snafu would you mind rebasing your PR? :)
Then I should be able to review and ask other coworkers to have a look

**Comment by skjnldsv** [comment]
Closing this pull request due to lack of recent activity and updates. We appreciate your contribution and encourage you to reopen or provide further updates if necessary. :relaxed: 
Our aim is to keep the project moving forward with active collaboration. Thank you for your understanding and continued support! :pray:

**Comment by Snafu** [comment]
Hi,
this pull request was no longer relevant because the strpos() calls were refactored to str_contains() in the following commit.

Author: Faraz Samapoor <f.samapoor@gmail.com>
Date: 02.06.2023 18:24:49
Commit hash: cfb921b26c656352f727b3903b8961e1c48a8051

Refactors "strpos" calls in /apps/files_external to improve code readability.

Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

**Pull request closed without merging**